pkg/reconciler/revision/revision.go

```diff
@@ -71,12 +71,17 @@ var (
 )
 
 func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) (bool, error) {
+	totalNumOfContainers := len(rev.Spec.Containers) + len(rev.Spec.InitContainers)
 	if rev.Status.ContainerStatuses == nil {
 		rev.Status.ContainerStatuses = make([]v1.ContainerStatus, 0, len(rev.Spec.Containers))
 	}
+	if rev.Status.InitContainerStatuses == nil {
+		rev.Status.InitContainerStatuses = make([]v1.ContainerStatus, 0, len(rev.Spec.InitContainers))
+	}
```

Comment:
I can't for the life of me figure out why we bother to initialise these arrays only to override them straight away 🤔 I know this is just doing what the pre-existing code does, but I'm almost sure we can drop the whole block

Reply:
Left over will omit.